web/src/message_edit.ts

```diff
@@ -965,7 +965,7 @@ export function try_save_inline_topic_edit($row: JQuery): void {
     const old_topic = message.topic;
     const new_topic = $row.find<HTMLInputElement>("input.inline_topic_edit").val();
     assert(new_topic !== undefined);
-    const topic_changed = new_topic !== old_topic && new_topic.trim() !== "";
+    const topic_changed = new_topic !== old_topic;
```

Comment:
We might want to `.trim` both sides of this comparison, since that's how topics are compared in the server?

Reply:
Maybe this? Totally untested, so not adding it myself.

``` diff
diff --git a/web/src/message_edit.ts b/web/src/message_edit.ts
index 4570f6ee39..9e1e0357f6 100644
--- a/web/src/message_edit.ts
+++ b/web/src/message_edit.ts
@@ -963,7 +963,7 @@ export function try_save_inline_topic_edit($row: JQuery): void {
     const message = message_lists.current.get(message_id);
     assert(message?.type === "stream");
     const old_topic = message.topic;
-    const new_topic = $row.find<HTMLInputElement>("input.inline_topic_edit").val();
+    const new_topic = $row.find<HTMLInputElement>("input.inline_topic_edit").val()?.trim();
     assert(new_topic !== undefined);
     const topic_changed = new_topic !== old_topic;
 
```

Reply:
Added a prep commit (3rd commit) for this as it solves some existing bugs independent to this feature.